File: problems/arrays-and-hashing/5-longest-common-prefix/rust/src/main.rs
fn longest_common_prefix(strs: Vec<String>) -> String {
    let mut prefix = String::new();

    for (i, c) in strs[0].chars().enumerate() {
        for str in &strs {
            if str.chars().nth(i) != Some(c) {
                return prefix
            }
        }
        prefix.push(c);
    }

    prefix
}

fn main() {
    let strs_1 = vec![String::from("bat"), String::from("bag"), String::from("bank"), String::from("band")];
    let prefix_1 = String::from("ba");
    let strs_2 = vec![String::from("dance"), String::from("dag"), String::from("danger"), String::from("damage")];
    let prefix_2 = String::from("da");
    let strs_3 = vec![String::from("neet"), String::from("feet")];
    let prefix_3 = String::from("");

    let res_1 = longest_common_prefix(strs_1);
    let res_2 = longest_common_prefix(strs_2);
    let res_3 = longest_common_prefix(strs_3);

    if res_1 != prefix_1 || res_2 != prefix_2 || res_3 != prefix_3 {
        println!("Actual results not equal with expected results.");
    }
    else {
        println!("Results correct.");
    }
}

File: problems/arrays-and-hashing/5-longest-common-prefix/cpp/main.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

string longestCommonPrefix(vector<string>& strs) {
    string firstStr = strs[0];
    string prefix = "";
    for (int i = 0; i < firstStr.length(); ++i) {
        char c = firstStr[i];
        for (auto str : strs) {
            if (i >= str.length() || str[i] != c) {
                return prefix;
            }
        }
        prefix.push_back(c);
    }
    return prefix;
}

int main() {
    vector<string> strs1 = { "bat", "bag", "bank", "band" };
    string prefix1 = "ba";
    vector<string> strs2 = { "dance", "dag", "danger", "damage" };
    string prefix2 = "da";
    vector<string> strs3 = { "neet", "feet" };
    string prefix3 = "";

    string res1 = longestCommonPrefix(strs1);
    string res2 = longestCommonPrefix(strs2);
    string res3 = longestCommonPrefix(strs3);

    if (res1 != prefix1 || res2 != prefix2 || res3 != prefix3) {
        cout << "Actual results not equal with expected results." << endl;
    }
    else {
        cout << "Results correct." << endl;
    }

    return 0;
}